internal/am/delete_transfer.go

```diff
@@ -0,0 +1,38 @@
+package am
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+
+	"github.com/artefactual-sdps/enduro/internal/sftp"
+)
+
+const DeleteTransferActivityName = "DeleteTransferActivity"
+
+type DeleteTransferActivityParams struct {
+	Destination string
+}
+
+type DeleteTransferActivity struct {
+	client sftp.Client
+	logger logr.Logger
+}
+
+func NewDeleteTransferActivity(logger logr.Logger, client sftp.Client) *DeleteTransferActivity {
+	return &DeleteTransferActivity{client: client, logger: logger}
+}
+
+func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) error {
+	a.logger.V(1).Info("Execute DeleteTransferActivity",
+		"Destination", params.Destination,
+	)
+
+	err := a.client.Delete(ctx, params.Destination)
+	if err != nil {
+		return fmt.Errorf("delete transfer: path: %s. error: %v", params.Destination, err)
```

Comment:
I'm suggesting a couple of changes: use `%q` and update wording to avoid using the word `error` (seems redundant and inconsistent with how we've been building error messages).

```suggestion
		return fmt.Errorf("delete transfer: path: %q: %v", params.Destination, err)
```